lib/VCE/Services/Switch.pm

```
@@ -377,7 +377,22 @@ sub _add_switch {
         return;
     }
 
-    return { results => [ { id => $id } ] };
+
+
+    #build rabbitmq client
+    my $client = GRNOC::RabbitMQ::Client->new(
+        user     => $self->rabbit_mq->{'user'},
+        pass     => $self->rabbit_mq->{'pass'},
+        host     => $self->rabbit_mq->{'host'},
+        timeout  => 30,
+        port     => $self->rabbit_mq->{'port'},
+        exchange => 'VCE',
+        topic    => 'VCE'
+	);    
+
+    my $res = $client->add_switch(switch_id => $id);
+    
+    return { results => [ { id => $id, status => $res } ] };
```

I'd prefer to return success only if the db insert is successful and the child process is created. Please wrap the add_switch call inside a db transaction and rollback if `$client->add_switch` fails.

---

I don't think we really want to remove the switch from the DB when we fail to launch a new process... instead I think we should return an error text with the id and status of 0 when it fails